### PR TITLE
Fixed column causes editors to be obscured

### PR DESCRIPTION
**Pull request by sonny-chan**
## Description
Locked columns can sometimes cause editors to appear behind the grid

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/adazzle/react-data-grid/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)
Locked columns will sometimes cause editors to appear behind the grid

**What is the new behavior?**
Editors opened in locked columns should appear normally.

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: packages/react-data-grid/src/Cell.js [hunk 1]
```diff
@@ -297,6 +297,18 @@ const Cell = React.createClass({
     }
   },
 
+  removeScroll() {
+    let ctrl: any = this; // flow on windows has an outdated react declaration, once that gets updated, we can remove this
+    if (ctrl.isMounted()) {
```
Comment (amanmahajan7) [inline packages/react-data-grid/src/Cell.js]: This is an anti pattern, any reason for using it?
https://facebook.github.io/react/blog/2015/12/16/ismounted-antipattern.html

File: packages/react-data-grid/src/Cell.js [hunk 2]
```diff
@@ -297,6 +297,18 @@ const Cell = React.createClass({
...
+      let node = ReactDOM.findDOMNode(this);
```
Comment (amanmahajan7) [inline packages/react-data-grid/src/Cell.js]: We should add a ref on the container [element](https://github.com/adazzle/react-data-grid/blob/master/packages/react-data-grid/src/Cell.js#L502)

https://facebook.github.io/react/docs/refs-and-the-dom.html
"React will call the ref callback with the DOM element when the component mounts, and call it with null when it unmounts."
In theory, this should eliminate the need to check isMounted.

File: packages/react-data-grid/src/Cell.js [hunk 3]
```diff
@@ -297,6 +303,14 @@ const Cell = React.createClass({
     }
   },
 
+  removeScroll() {
+    let node = ReactDOM.findDOMNode(this);
```
Comment (diogofcunha) [inline packages/react-data-grid/src/Cell.js]: 😱 isn't there another way to do this? when using react mutating the DOM directly is dangerous and not adviced at all

Comment (sonny-chan) [inline packages/react-data-grid/src/Cell.js]: @diogofcunha i'm not sure of another way to do this. if you have any good ideas, please let me know.

I was only following the pattern used in setScrollLeft: https://github.com/adazzle/react-data-grid/blob/9d0e0bff012f9cdedf21c87a19314d2e14bee2e1/packages/react-data-grid-addons/src/draggable/DraggableHeaderCell.js#L18

**Pull request merged**